webapp/src/components/App/Singlestudy/explore/common/OutputFilters.tsx

```diff
@@ -0,0 +1,64 @@
+import { FieldPath } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import SelectFE from "../../../../common/fieldEditors/SelectFE";
+import Fieldset from "../../../../common/Fieldset";
+import { ControlPlus } from "../../../../common/Form";
+
+type FilteringType = "hourly" | "daily" | "weekly" | "monthly" | "annual";
+
+interface FilterFieldValues {
+  filterSynthesis: Array<FilteringType>;
+  filterByYear: Array<FilteringType>;
+}
+
+interface Props<T extends FilterFieldValues> {
+  handleAutoSubmit: (name: string, selection: string[]) => void;
+  control: ControlPlus<T>;
+}
+
+function OutputFilters<T extends FilterFieldValues>(props: Props<T>) {
+  const { handleAutoSubmit, control } = props;
+  const [t] = useTranslation();
+
+  const filterOptions = ["hourly", "daily", "weekly", "monthly", "annual"].map(
+    (item) => ({
+      label: t(`global.time.${item}`),
+      value: item,
+    })
+  );
+
+  const renderFilter = (filterName: FieldPath<T>) => (
```

Comment:
`FieldPath<FilterFieldValues>` because you only support two keys here.

Reply:
TS not happy about it